geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala

```diff
@@ -214,7 +214,11 @@ class AccumuloDataStore(val connector: Connector,
 
     // Prevent modifying wrong type if type names don't match
     if (!schemaTypeName.equals(typeName.toString)) {
-      throw new UnsupportedOperationException("Updating the type name of a schema is not allowed " + schemaTypeName + " " + typeName)
+      throw new IllegalArgumentException(s"Provided type name $typeName does not match schema type name $schemaTypeName")
```

Comment:
not a big deal, and I know @Afterglow375 just said to change it, but i think UnsupportedOperationException is actually correct here - theoretically the geotools API we implement would allow you to do this, but we don't have it implemented.